src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -342,6 +343,27 @@ long MacroAssembler::get_const(address a) {
   }
   return (long) x;
 }
+// Branch-free implementation to convert !0 to false
+// Set register dst to true if dst is non zero using temp for calculations on Power Version<10.
+// Set register dst to true if dst is non zero for Power 10 and above machines.
+void MacroAssembler::normalize_bool(Register dst, Register temp, bool use_64bit) {
+
+  if (VM_Version::has_brw()) {
+    if(use_64bit)
+      cmpdi(CCR0, dst, 0);
+    else
+      cmpwi(CCR0, dst, 0);
+    setbcr(dst, CCR0, Assembler::zero);
```

Comment:
This is what I understood after implementation & definition: 
```
If bit BI of the CR contains a 1, register RT is set to 0. Otherwise, register RT is set to 1.
```
CCR0 will contain `1` when `dst == 0`. then  `dst` will be set to `1` by `setbcr`.

Reply:
I'd prefer using `Assembler::equal` because we have compared with 0 before.